include/array/array.h

```diff
@@ -2971,7 +2973,7 @@ const_array_ref<U, Shape> reinterpret(const array<T, Shape, Alloc>& a) {
  * type `U` using `const_cast`. */
 template <class U, class T, class Shape>
 array_ref<U, Shape> reinterpret_const(const const_array_ref<T, Shape>& a) {
-  return array_ref<U, Shape>(const_cast<U*>(a.base()), a.shape());
+  return array_ref<U, Shape>(const_cast<U*>(a.base()), a.shape(), std::false_type{});
```

Comment:
`const_array_ref<T, ...>` is just `array_ref<const T, ...>` so it would just work if `T` was const. That said: I'm generally avoiding using helpers if they are trivial because of compile time and debug build performance concerns. At one point working on this library, build times were pretty bad. I worked to reduce the pre-optimization code size, which made a good dent in the problem.

I'm going to replace `std::false_type` with something more descriptive though.